src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -50,11 +51,25 @@ public Publish(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFac
         public async Task<IEnumerable<PublishResult>> ProcessAsync(string folderPath)
         {
             this.publishResults.Clear();
+
             if (!this.IsValidInput(folderPath))
             {
                 return this.publishResults;
             }
 
+            var productValidate = new ValidateProduct();
+            var errors = productValidate.IsProductValid(folderPath);
```

Comment:
If you validate all products, then use plural form in method name.

Reply:
Done, but still product or products depend on the user yaml file.